addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3234,6 +3235,22 @@ export class OdooEditor extends EventTarget {
                 fragment.appendChild(p);
             }
         }
+
+        // Split elements containing <br> into seperate elements for each line.
+        const brs = fragment.querySelectorAll('br');
+        for (const br of brs) {
+            const block = closestBlock(br);
+            if (
+                ['P', 'H1', 'H2', 'H3', 'H4', 'H5', 'H6'].includes(block.nodeName) &&
+                !block.closest('li')
+            ) {
+                const isEmptyLineBreak = block.firstChild.nodeName === 'BR';
+                const splitElement = splitAroundUntil(br, block);
+                if (isEmptyBlock(splitElement) && !isEmptyLineBreak) {
+                    splitElement.remove();
+                }
```

Comment:
I don't understand this condition nor the case it wants to cover.

Reply:
Let's assume I have `<p>abc<br><br>def</p>`. The first `<br>` serves as a line break, while the second `<br>` indicates an empty line.

If I split around the `<p>` tag at the first `<br>`, it would become `<p>abc</p><p><br></p><p><br>def<p>`. I need to remove the `<p>` element containing the first `<br>`. However, for the second `<br>` I don't want to remove the split element, as it signifies an empty line. And how can I determine if a `<br>` represents an empty line, One way is to check if its closest block's `firstChild` is `<br>`.

See this test case: https://github.com/odoo/odoo/pull/136743/files#diff-ed334b1f9f243d6fb9d00527d62af2dd515fd4ea0442c9c89d811c0caa1ad19bR842-R848

Reply:
I see, thanks. I suggest to add a comment on top of this code then, explaining this, as it's impossible to figure out without a deep knowledge of what is going on... ^^'

Reply:
Added the comment. Also, I realized if I `splitAroundUntil` br till its `closestBlock` the splitElement will always be an emptyBlock :eyes: so I actually did not need to check that, Therefore I changed the if condition from `if (isEmptyBlock(splitElement) && !isEmptyLineBreak)` to `if (!isEmptyLineBreak)`

Reply:
Thanks! Now that I have a better understanding of the issue, I suggest this:
```js
                // A linebreak at the beginning of a block is an empty line.
                const isEmptyLine = block.firstChild.nodeName === 'BR';
                // Split blocks around it until only the BR remains in the block.
                const remainingBrContainer = splitAroundUntil(br, block);
                // Remove the container unless it represented an empty line.
                if (!isEmptyLine) {
                    remainingBrContainer.remove();
                }
```